pkg/pgmodel/model/series.go

```diff
@@ -41,14 +41,12 @@ type Series struct {
 	epoch    SeriesEpoch
 
 	metricName string
-	str        string
 }
 
-func NewSeries(key string, labelPairs []prompb.Label) *Series {
+func NewSeries(labelPairs []prompb.Label) *Series {
 	series := &Series{
 		names:    make([]string, len(labelPairs)),
 		values:   make([]string, len(labelPairs)),
```

Comment:
May be adding separate dictionary for keys & values would reduce size further and the same encoding could be used here in names and values instead of string[]?